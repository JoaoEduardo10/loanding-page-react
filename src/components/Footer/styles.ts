import styled, { css } from 'styled-components';

import { Conteiner as TextComponent } from '../TextComponent/styles';
import { Conteiner as SectionConteiner } from '../SectionConteiner/styles';

export const Conteiner = styled.div`
    ${({ theme }) => css`
        text-align: center;
        border-top: 0.1rem solid ${theme.colors.mediumGray};
        
        a {
            text-decoration: none;
            color: inherit;
        }

        & ${TextComponent} {
            font-size: ${theme.font.size.small};
        }

        & ${SectionConteiner} {
            padding-top: 0;
            padding-bottom: 0;
        }
    `}
`;