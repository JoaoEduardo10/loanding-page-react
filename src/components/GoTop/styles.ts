import styled, { css } from 'styled-components';

export const Conteiner = styled.a`
    ${({ theme }) => css`
        position: fixed;
        background: ${theme.colors.primaryColor};
        display: flex;
        align-items: center;
        justify-content: center;
        width: 4rem;
        height: 4rem;
        bottom: 2rem;
        right: 2rem;
        z-index: 6;
        opacity: 0.7;
    `}
`;