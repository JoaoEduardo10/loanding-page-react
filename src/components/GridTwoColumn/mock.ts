type mockGridProps = {
    title: string
    text: string
    srcImg: string
}

export const mockGrid: mockGridProps = {
	text: 'Lorem ipsum, dolor sit amet consectetur adipisicing elit. Illum maxime distinctio illo pariatur incidunt omnis, molestias, dolores facere qui autem eos laboriosam quam voluptatibus placeat itaque, dolore similique alias perspiciatis!',
	title: 'GridTwoColumn',
	srcImg: 'assets/images/javascript.svg'
};