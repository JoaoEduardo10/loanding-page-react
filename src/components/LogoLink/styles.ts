import styled, { css } from 'styled-components';

export const Conteiner = styled.a`
    ${() => css`
        display: flex;
        align-items: center;
        text-decoration: none;
        color: inherit;

        > img {
            max-height: 3rem;
        }
    `}
`;