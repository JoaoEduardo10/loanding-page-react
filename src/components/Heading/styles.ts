import styled, { css } from 'styled-components';
import { ThemeProps } from '../../styles/theme';

type Props = 'big' | 'small' | 'medium' | 'huge'


const tileSize = {
	small: (theme: ThemeProps) => css`
        font-size: ${theme.font.size.medium};
    `,
	big: (theme: ThemeProps) => css`
        font-size: ${theme.font.size.xlarge};
    `,
	medium: (theme: ThemeProps) => css`
    font-size: ${theme.font.size.large};
    `,
	huge: (theme: ThemeProps) => css`
        font-size: ${theme.font.size.xhuge};
        ${titleMedia(theme)}
    `,
};

const titleMedia = (theme: ThemeProps) => css`
    @media ${theme.media.lteMedia} {
        font-size: ${theme.font.size.xlarge};
    }
`;

const titleUppercase = (uppercase: boolean) => css`
    text-transform: ${uppercase ? 'uppercase' : 'none'};
`;

export const Title = styled.h1<{ darkColor: boolean, size: Props, uppercase: boolean }>`
    ${({ theme, darkColor, size, uppercase }) => css`
        color: ${ darkColor ? theme.colors.white : theme.colors.primaryColor};
        ${tileSize[size](theme)}
        ${titleUppercase(uppercase)}
    `}
`;