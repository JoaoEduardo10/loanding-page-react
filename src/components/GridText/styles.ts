import styled, { css } from 'styled-components';

import { Title as HeadingConteiner } from '../Heading/styles';
import { Conteiner as TextComponent } from '../TextComponent/styles';

export const Conteiner = styled.div`
    ${({ theme }) => css`
        
        ${TextComponent} {
            margin-bottom: ${theme.spacings.xhuge};
        }

    `}
`;

export const Grid = styled.div`
    ${({ theme }) => css`
        counter-reset: grid-counter;
        display: grid;
        grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
        gap: ${theme.spacings.large};
    `}
`;

export const GridElement = styled.div`
    ${({ theme }) => css`
        ${HeadingConteiner} {
            position: relative;
            left: 5rem;
        }

        ${HeadingConteiner}::before{
            position: absolute;
            counter-increment: grid-counter;
            content: counter(grid-counter);
            font-size: ${theme.font.size.xhuge};
            left: -5rem;
            top: -3rem;
            transform: rotate(5deg);
        }
    `}
`;