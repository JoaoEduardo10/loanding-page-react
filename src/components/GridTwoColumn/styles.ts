import styled, { css } from 'styled-components';
import { Title as Heading } from '../Heading/styles';

export const Conteiner = styled.div`
    ${({ theme }) => css`
        display: grid;
        grid-template-columns: 1fr 2fr;
        align-items: center;
        gap: ${theme.spacings.large};

        @media ${theme.media.lteMedia} {
            grid-template-columns: 1fr;
            text-align: center;
        }

        & ${Heading} {
            margin-bottom:  ${theme.spacings.xlarge};
        }
    `}
`;

export const TextConteiner = styled.div`
    ${({ theme }) => css`
        @media ${theme.media.lteMedia} {
            margin-bottom: ${theme.spacings.large};
        }
    `}
`;

export const ImageConteiner = styled.div`
    ${() => css``}
`;

export const Image = styled.img`
    ${() => css``}
`;