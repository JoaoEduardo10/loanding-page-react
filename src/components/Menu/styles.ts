import styled, { css } from 'styled-components';

import { Conteiner as SectionConteiner } from '../SectionConteiner/styles';
import { Title as Heading } from '../Heading/styles';

const menuVisible = () => css`
    visibility: visible;
    opacity: 1;
`;

export const Conteiner = styled.aside<{ visible: boolean }>`
    ${({ theme, visible }) => css`
        position: fixed;
        top: 0;
        left: 0;
        bottom: 0;
        right: 0;
        width: 100%;
        height: 100%;
        border-bottom: 0.1rem solid ${theme.colors.mediumGray};
        background-color: ${theme.colors.white};
        overflow-y: auto;
        transition: all 300ms ease-in-out;

        > ${SectionConteiner} {
            padding-top: 0;
            padding-bottom: 0;
        }

        & ${Heading} {
            margin-top: 0;
            margin-bottom: 0%;
        }

        @media ${theme.media.lteMedia} {
            height: 100vh;
            visibility: hidden;
            opacity: 0;
            ${visible && menuVisible()}

            ${SectionConteiner} {
                display: grid;
                grid-template-columns: 1fr;
                grid-template-rows: 1fr;
                height: 100vh;
                align-items: center;
            }

            & ${Heading} {
                display: flex;
                justify-content: center;
                margin-bottom: ${theme.spacings.large};
            }
        }
    `}
`;

export const MenuConteiner = styled.div`
    ${({ theme }) => css`
        display: flex;
        justify-content: space-between;
        align-items: center;

        @media ${theme.media.lteMedia} {
            display: block;
            text-align: center;
            padding: ${theme.spacings.xxlarge} 0;
        }
    `}
`;

export const Button = styled.button`
    ${({ theme }) => css`
        z-index: 6;
        position: fixed;
        top: 2rem;
        right: 2rem;
        width: 4rem;
        height: 4rem;
        background-color: ${theme.colors.primaryColor};
        color: ${theme.colors.white};
        border: none;
        display: none;
        cursor: pointer;

        @media ${theme.media.lteMedia} {
            display: flex;
            align-items: center;
            justify-content: center;
        }

        > svg {
            width: 2.5rem;
            height: 2.5rem  
        }
    `}
`;