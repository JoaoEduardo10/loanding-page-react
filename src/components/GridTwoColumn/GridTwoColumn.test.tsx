import { GridTwoColumn } from '.';
import { theme } from '../../styles/theme';
import { renderTheme } from '../../utils/renderTheme';

import { mockGrid } from './mock';

describe('<GridTwoColumn />', () => {
	it('should render GridTwoColumn with background white', () => {
		const { container } = renderTheme(<GridTwoColumn {...mockGrid} />);

		expect(container.firstChild).toHaveStyleRule('background-color', theme.colors.white);

		expect(container.firstChild).toMatchSnapshot();
	});

	it('should render GridTwoColumn with background black', () => {
		const { container } = renderTheme(<GridTwoColumn background {...mockGrid} />);

		expect(container.firstChild).toHaveStyleRule('background-color', theme.colors.primaryColor);

		expect(container.firstChild).toMatchSnapshot();

	});
});