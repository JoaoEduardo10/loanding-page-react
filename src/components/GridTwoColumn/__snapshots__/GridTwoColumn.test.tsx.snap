// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GridTwoColumn /> should render GridTwoColumn with background black 1`] = `
.c5 {
  color: #fff;
  font-size: 4.0rem;
  text-transform: none;
}

.c1 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
}

.c0 {
  background-color: #fff;
  color: #0A1128;
  background-color: #0A1128;
  color: #fff;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c6 {
  font-size: 2.4rem;
}

.c2 {
  display: grid;
  grid-template-columns: 1fr 2fr;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 3.2rem;
}

.c2 .c4 {
  margin-bottom: 4.0rem;
}

@media (max-width:768px) {
  .c2 {
    grid-template-columns: 1fr;
    text-align: center;
  }
}

@media (max-width:768px) {
  .c3 {
    margin-bottom: 3.2rem;
  }
}

<section
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <h1
          class="c4 c5"
        >
          GridTwoColumn
        </h1>
        <p
          class="c6"
        >
          Lorem ipsum, dolor sit amet consectetur adipisicing elit. Illum maxime distinctio illo pariatur incidunt omnis, molestias, dolores facere qui autem eos laboriosam quam voluptatibus placeat itaque, dolore similique alias perspiciatis!
        </p>
      </div>
      <div
        class=""
      >
        <img
          alt="GridTwoColumn"
          class=""
          src="assets/images/javascript.svg"
        />
      </div>
    </div>
  </div>
</section>
`;

exports[`<GridTwoColumn /> should render GridTwoColumn with background white 1`] = `
.c5 {
  color: #0A1128;
  font-size: 4.0rem;
  text-transform: none;
}

.c1 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
}

.c0 {
  background-color: #fff;
  color: #0A1128;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c6 {
  font-size: 2.4rem;
}

.c2 {
  display: grid;
  grid-template-columns: 1fr 2fr;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 3.2rem;
}

.c2 .c4 {
  margin-bottom: 4.0rem;
}

@media (max-width:768px) {
  .c2 {
    grid-template-columns: 1fr;
    text-align: center;
  }
}

@media (max-width:768px) {
  .c3 {
    margin-bottom: 3.2rem;
  }
}

<section
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <h1
          class="c4 c5"
        >
          GridTwoColumn
        </h1>
        <p
          class="c6"
        >
          Lorem ipsum, dolor sit amet consectetur adipisicing elit. Illum maxime distinctio illo pariatur incidunt omnis, molestias, dolores facere qui autem eos laboriosam quam voluptatibus placeat itaque, dolore similique alias perspiciatis!
        </p>
      </div>
      <div
        class=""
      >
        <img
          alt="GridTwoColumn"
          class=""
          src="assets/images/javascript.svg"
        />
      </div>
    </div>
  </div>
</section>
`;
